### PR TITLE
Basic DataFusion Select Functionality

### PR DESCRIPTION
**Pull request by jdye64**
Basic SELECT functionality working, PyTests that are not working due to other issues are referenced in the `@pytest.mark.skip(...)`

This closes: #468 
This closes: #490 
This closes: #496 
This closes: #483

### CODE REVIEW

**Comment by jdye64** [comment]
It will be much easier to review this PR after [471](https://github.com/dask-contrib/dask-sql/pull/471) is merged. Feel free to wait until after that for a review.

File: dask_planner/src/sql/types.rs
Comment (ayushdg) [inline dask_planner/src/sql/types.rs:116]: Do we want to pass the same timezone even when the type is a `timestamp` type without a timezone?

File: dask_sql/mappings.py
Comment (ayushdg) [inline dask_sql/mappings.py:132]: Some of these earlier assumptions about calcite may not hold true for datafusion and will need updating as and when mappings are encountered

File: dask_sql/mappings.py
Comment (ayushdg) [inline dask_sql/mappings.py:264]: Debug print

File: dask_sql/physical/rel/logical/project.py
Comment (ayushdg) [inline dask_sql/physical/rel/logical/project.py:88]: Same as above

File: dask_sql/mappings.py
Comment (jdye64) [reply dask_sql/mappings.py:132]: Completely agree. Even though all the current tests are passing I am sure there are some missing edge cases. I was thinking about opening a issue just for that alone and was hoping that you might be able to help with that one since you have done a lot of work in the date/time area. Could I open an issue and assign it to you for a subsequent PR?

File: dask_planner/src/sql/types.rs
Comment (jdye64) [reply dask_planner/src/sql/types.rs:116]: I honestly don't know? You would know much better. Which way is the correct way?

Comment (ayushdg) [reply dask_planner/src/sql/types.rs:116]: A default timestamp (without timezone) based on postresql is interpreted as the timezone in utc irrespective of the offset provided. So in this case the tz should always be utc.

Comment (jdye64) [reply dask_planner/src/sql/types.rs]: That might have been handled by the library but either way I added a check just to be sure.

File: continuous_integration/recipe/meta.yaml
Comment (ayushdg) [inline continuous_integration/recipe/meta.yaml:33]: Based on the discussion [here](https://github.com/dask-contrib/dask-sql/pull/471#discussion_r853217325) is this needed?

File: continuous_integration/recipe/run_test.py
Comment (ayushdg) [inline continuous_integration/recipe/run_test.py:17]: Out of curiosity what functionality is missing for this to work?

File: dask_planner/src/expression.rs
Comment (ayushdg) [inline dask_planner/src/expression.rs:101]: If the debug prints are useful as work progresses, it might be worth adding a `TODO` comment to clean up later

File: dask_sql/context.py
Comment (ayushdg) [inline dask_sql/context.py:813]: If one doesn't exist already we should add a test case checking for this.

File: tests/unit/test_mapping.py
Comment (ayushdg) [inline tests/unit/test_mapping.py:24]: Is the default datafusion unit in microseconds?

File: .github/workflows/test.yml
Comment (ayushdg) [inline .github/workflows/test.yml:165]: is the pyarrow requirement no longer needed?

**Review by ayushdg — CHANGES_REQUESTED** [review]
Few minor questions but generally things look good to me

File: continuous_integration/recipe/meta.yaml
Comment (jdye64) [reply continuous_integration/recipe/meta.yaml:33]: No, it is not needed. I had based this PR off the `datafusion-aggregate` one and probably did that before removing this in the other PR. It can be removed and I will remove it.

File: dask_planner/src/expression.rs
Comment (jdye64) [reply dask_planner/src/expression.rs:101]: Yep, my plan is to not merge with upstream master during all of these commits. What that will give us is basically another shot to review before we merge into main as all of these type changes will stick out like red flags and can be changed then but at the same time can be left in for now to assist with current development work.

Long story short. 100% plan on removing them during that final PR

File: dask_sql/context.py
Comment (jdye64) [reply dask_sql/context.py:813]: Several already exist one coming top of mind is the `test_wrong_input` pytest in `test_select.py`

File: tests/unit/test_mapping.py
Comment (jdye64) [reply tests/unit/test_mapping.py:24]: Yes, however that can be changed and I should look into finding out how I can make it default to milliseconds instead

File: .github/workflows/test.yml
Comment (jdye64) [reply .github/workflows/test.yml:165]: Thankfully no. I found a portion of the DataFusion library where those mappings were already present so I just wrote a Rust function that I exposed to Python that allowed me to use that instead of including PyArrow in the Python layer since DataFusion already includes Arrow.

Previously I didn't know that portion of the DataFusion library existed so that was the reason for the original inclusion of PyArrow.

File: continuous_integration/recipe/run_test.py
Comment (jdye64) [reply continuous_integration/recipe/run_test.py:17]: The `SUM()` operation has not yet been implemented as part of the `RexCall -> call.py` port

File: tests/unit/test_mapping.py
Comment (ayushdg) [reply tests/unit/test_mapping.py:24]: Actually, even if it needs a few changes on the python side my personal preference is to staying with this default and evaluating if we need to move to millis if it causes problems down the line

**Review by ayushdg — APPROVED** [review]
Changes lgtm! Thanks a lot @jdye64

**Pull request merged**
